private/buf/bufcli/bufcli.go

```diff
@@ -536,22 +538,20 @@ func NewContextModifierProvider(
 	container appflag.Container,
 ) func(string) (func(context.Context) context.Context, error) {
 	return func(address string) (func(context.Context) context.Context, error) {
-		machine, err := netrc.GetMachineForName(container, address)
-		if err != nil {
-			return nil, fmt.Errorf("failed to read server password from netrc: %w", err)
-		}
-		var password string
-		if machine != nil {
-			password = machine.Password()
+		token := container.Env(tokenEnvKey)
+		if token == "" {
+			machine, err := netrc.GetMachineForName(container, address)
+			if err != nil {
+				return nil, fmt.Errorf("failed to read server password from netrc: %w", err)
+			}
+			token = machine.Password()
 		}
 		return func(ctx context.Context) context.Context {
-			return rpcauth.WithToken(
```

Comment:
We do ignore empty values in `WithToken`, but this is making it a bit more explicit